### PR TITLE
Move hdf5 dependencies from adjoint to hdf5 application

### PR DESCRIPTION
**Pull request by msandre**
- Removed hdf5 compile time dependencies from AdjointFluidApplication.

- Added hdf5 non historical nodal value io to support features removed from AdjointFluidApplication (@qaumann let me know if this creates a problem for you)

- Refactored hdf5 process construction and added primal io scripts.

Adjoint and HDF5 tests pass.

### CODE REVIEW

File: applications/HDF5Application/custom_io/hdf5_nodal_solution_step_bossak_io.cpp
Comment (jcotela) [inline applications/HDF5Application/custom_io/hdf5_nodal_solution_step_bossak_io.cpp]: You should be able to do `rVariable == ACCELERATION`, which is a key comparison, and likely cheaper than a string comparison.

**Review by jcotela — DISMISSED** [review]
:+1: approving

**Review by jcotela — APPROVED** [review]

**Comment by msandre** [comment]
Thanks, I'll wait until next week to merge in case @qaumann has an issue with it.

**Pull request merged**
